GrandCentralBoard/Widgets/Bonus/Extensions/Array.swift

```diff
@@ -18,3 +18,17 @@ extension RangeReplaceableCollectionType where Generator.Element : Equatable {
     }
 
 }
+
+extension SequenceType where Generator.Element == Bonus {
+    func sortByDate() -> [Bonus] {
+        return self.sort({ $0.date.compare($1.date) == .OrderedDescending })
+    }
+
+    func flatten() -> [Bonus] {
+        let mappedBonuses = self.map { bonus in
+            return [ Bonus(name: bonus.name, amount: bonus.amount, receiver: bonus.receiver, date: bonus.date, childBonuses: [])]
```

Comment:
I wanted just in case to prevent from further extraction of child bonuses. But maybe it's better not to remove data. 

Thanks for the info about `flatMap`. Yes, it can be simplified.